lwr_controllers/src/arm_state_controller.cpp

```diff
@@ -0,0 +1,121 @@
+        
+        // get publishing period
+        if (!nh_.getParam("publish_rate", publish_rate_)) {
+            ROS_ERROR("Parameter 'publish_rate' not set");
+            return false;
+        }
+        
+        realtime_pub_.reset(new realtime_tools::RealtimePublisher<lwr_controllers::ArmState>(nh_,"arm_state",4));
+        realtime_pub_->msg_.est_ext_torques.resize(kdl_chain_.getNrOfJoints());
+        
+        gravity_.reset(new KDL::Vector(0.0, 0.0, -9.81)); // TODO: compute from actual robot position (TF?)
+        id_solver_.reset(new KDL::ChainIdSolver_RNE(kdl_chain_, *gravity_));
+        jac_solver_.reset(new KDL::ChainJntToJacSolver(kdl_chain_));
+        fk_solver_.reset(new KDL::ChainFkSolverPos_recursive(kdl_chain_));
+        jacobian_.reset(new KDL::Jacobian(kdl_chain_.getNrOfJoints()));
+        joint_position_.reset(new KDL::JntArray(kdl_chain_.getNrOfJoints()));
+        joint_velocity_.reset(new KDL::JntArray(kdl_chain_.getNrOfJoints()));
+        joint_acceleration_.reset(new KDL::JntArray(kdl_chain_.getNrOfJoints()));
+        joint_wrenches_.reset(new KDL::Wrenches(kdl_chain_.getNrOfJoints()));
+        joint_effort_est_.reset(new KDL::JntArray(kdl_chain_.getNrOfJoints()));
+        
+        return true;		
+    }
+    
+    void ArmStateController::starting(const ros::Time& time)
+    {
+        last_publish_time_ = time;
+        for (unsigned i = 0; i < joint_handles_.size(); i++){
+            (*joint_position_)(i) = joint_handles_[i].getPosition();
+            (*joint_velocity_)(i) = joint_handles_[i].getVelocity();
+            (*joint_acceleration_)(i) = 0; 
+        }
+        ROS_INFO("started");
+    }
+    
+    void ArmStateController::update(const ros::Time& time, const ros::Duration& period)
+    {
+        // limit rate of publishing
+        if (publish_rate_ > 0.0 && last_publish_time_ + ros::Duration(1.0/publish_rate_) < time){
+            
+            if (realtime_pub_->trylock()) {
+                // we're actually publishing, so increment time
+                last_publish_time_ = last_publish_time_ + ros::Duration(1.0/publish_rate_);
+                
+                for (unsigned i = 0; i < joint_handles_.size(); i++) {
+                    float acceleration = filters::exponentialSmoothing((joint_handles_[i].getPosition() - (*joint_position_)(i))/period.toSec(), joint_handles_[i].getVelocity(), 0.2);
+                    (*joint_position_)(i) = joint_handles_[i].getPosition();
+                    (*joint_velocity_)(i) = joint_handles_[i].getVelocity();
+                    (*joint_acceleration_)(i) = acceleration; 
+                }
+                                
+                // Compute Dynamics 
+                int ret = id_solver_->CartToJnt(*joint_position_, 
+                                                *joint_velocity_, 
+                                                *joint_acceleration_, 
+                                                *joint_wrenches_,
```

Comment:
No, it's fine, I was just wondering that if you never consider external forces (joint_wrenhes_ = 0), perhaps there was no need to use the solver, because you are not actually solving neither a FD nor an ID problem, also because the motion is measured not planned, but the numbers are the same as in an ID problem.

So, as you wish, no problem, let me know when to merge.

Though, I'd change the var name in case we ever use it, since it is not joint wrenches as if the joint generated the wrenches, but wrenches at the segments, I believe expressed in the corresponding joint frame from [this](https://github.com/orocos/orocos_kinematics_dynamics/blob/master/orocos_kdl/src/chainidsolver_recursive_newton_euler.cpp#L71), it doesn't say that, only:
`f_ext 	The external forces (no gravity) on the segments Output parameters:`
anyways, it is zero in our case.